R/pivot_wider.R

```diff
@@ -0,0 +1,65 @@
+#' Pivot data from long to wide
+#
+#' @description
+#'
+#' `pivot_wider()` "widens" data, increasing the number of columns and
+#' decreasing the number of rows. The inverse transformation is
+#' [pivot_longer()].
+#'
+#' @param data
+#' @param id_cols A set of columns that uniquely identifies each observation.
+#'   Defaults to all columns in `data` except for the columns specified in
+#'   `names_from` and `values_from`. Typically used when you have additional
+#'   variables that is directly related.
+#' @param names_from,values_from A pair of arguments describing which column
+#'   (or columns) to get the name of the output column (`name_from`), and
+#'   which column (or columns) to get the cell values from (`values_from`).
+#'
+#' @export
+#'
+#' @examples
+#' example_dt <- data.table(z = rep(c("a", "b", "c"), 2),
+#'                          stuff = c(rep("x", 3), rep("y", 3)),
+#'                          things = 1:6)
+#'
+#' example_dt %>%
+#'   dt_pivot_wider(names_from = stuff, values_from = things)
+dt_pivot_wider <- function(data,
+                           id_cols = NULL,
+                           names_from = name,
+                           names_sep = "_",
+                           values_from = value) {
+
```

Comment:
I think this function is particularly helpful as I don't think `dcast()` is intuitive. Love what you've done here.